pkgs/tools/misc/microplane/default.nix

```diff
@@ -0,0 +1,41 @@
+{ stdenv, buildGoPackage, fetchFromGitHub }:
+
+buildGoPackage rec {
+  pname = "microplane";
+  version = "0.0.25";
+
+  src = fetchFromGitHub {
+    owner = "Clever";
+    repo = "microplane";
+    rev = "v${version}";
+    sha256 = "0vynkv3d70q8d079kgdmwbavyyrjssqnd223dl1rikyy7sd5sli8";
+  };
+
+  goPackagePath = "github.com/Clever/microplane";
+
+  subPackages = ["."];
+
+  # Regenerate deps.nix with the following steps:
+  #
+  # git clone git@github.com:Clever/microplane.git
+  # cd microplane
+  # git checkout v<version>
+  # dep2nix
+
+  goDeps = ./deps.nix;
+
+  buildFlagsArray = ''
+    -ldflags="-s -w -X main.version=v${version}"
+  '';
+
+  postInstall = ''
+    mv $out/bin/microplane $out/bin/mp
```

Comment:
I am not a fan of putting a binary on every available two or three letter combination. The binary name is ``microplane`` and should stay like this. You can add such aliases to your .bashrc.

Reply:
Sounds good to me :+1: 

I was trying to stay close to upstream, and all of their documentation and blog articles I've seen refer to the binary as `mp`, so that's something to take in to account, but I think giving it its full name `microplane` won't be confusing. It will also prevent any unnecessary name conflicts too.

Reply:
If `mp` is such a recurring name in official documentations and tutorials, I suggest you create a symlink instead of moving. It will provide no surprises to new users.

Reply:
Ok, I did another quick check, and I can't find mention of upstream calling the binary by its full name. 

And here's some new pieces of info that may help bring consensus...

Looks like the binary refers to itself as `mp`:
```
$ microplane 
Microplane makes git changes across many repos

Usage:
  mp [command]

Available Commands:
  clone       Clone all repos targeted by init
  docs        Generates markdown docs for each command
  help        Help about any command
  init        Initialize a microplane workflow
  merge       Merge pushed changes
  plan        Plan changes by running a command against cloned repos
  push        Push planned changes
  status      Status shows a workflow's progress

Flags:
  -h, --help          help for mp
  -r, --repo string   single repo to operate on

Use "mp [command] --help" for more information about a command.
```

And it looks like upstream calls the binary the shorter name in [their releases](https://github.com/Clever/microplane/releases/tag/v0.0.25):
> mp-0.0.25-darwin-amd64
> mp-0.0.25-linux-amd64

Reply:
The symlink should suffice.

Reply:
Ok just symlinked `microplane` to `mp`.